src/Data/Auth/PrimaryAuthController.ts

```diff
@@ -0,0 +1,120 @@
+import Keycloak, { KeycloakConfig, KeycloakInitOptions } from "keycloak-js";
+import { AuthController } from "@/Core";
+import { removeCookie } from "../Common/CookieHelper";
+import keycloakConf from "./keycloak.json";
+
+interface AuthConfig extends KeycloakConfig {
+  method: "oidc";
+}
+
+interface LocalConfig {
+  method: "database";
+}
+
+/**
+ * PrimaryAuthController class.
+ * This class is responsible for handling authentication.
```

Comment:
could you add more about the fact it does handle both keycloak and localauth?

Reply:
added